src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -2707,6 +2707,12 @@ void Unit::_UpdateSpells(uint32 time)
         }
     }
 
+    for (auto nexctSpellCrit : m_spellCritMap)
+    {
+        if (!nexctSpellCrit.first.first)
```

Comment:
the first "first" gets me the pair.. and with the other first i'll get the Unit.. so i check if the Unit in the map is null or not.. if the Unit is null, i remove them..

Do you want to say i should comment this line?